src/components/application_manager/src/commands/mobile/system_request.cc

```diff
@@ -102,6 +102,54 @@ class QueryAppsDataValidator {
     return true;
   }
 
+  /**
+   * \brief Function checks, if language json array has 'ttsName' parameter omitted.
+   * \param languages SmartObject containing languages json array
+   * \return true of parameter was omitted, false otherwise
+  */
+  bool CheckIfHasOmittedParam(const smart_objects::SmartObject& languages) {
+    const size_t languages_array_size = languages.length();
+    for (size_t idx = 0; idx < languages_array_size; ++idx) {
+      const smart_objects::SmartObject& language = languages.getElement(idx);
+      if (smart_objects::SmartType_Map != language.getType()) {
+        LOG4CXX_DEBUG(logger_,
+                     kQueryAppsValidationFailedPrefix
+                         << "language is not a map.");
+        return false;
+      }
+      const std::string language_name = (*language.map_begin()).first;
+      if (!language[language_name].keyExists(json::ttsName) ||
+          language[language_name][json::ttsName].empty()) {
+        LOG4CXX_DEBUG(logger_,
+                     kQueryAppsValidationFailedPrefix
+                         << "'languages.ttsName' doesn't exist");
+        return true;
+      }
+    }
+    return false;
+  }
+
+  /**
+   * \brief Function writes 'app_id' value to ttsName parameter.
+   * \param languages SmartObject containing languages json array
+   * \param app_id string containt application id
+   * \return true of parameter was omitted, false otherwise
+  */
+  void WriteAppIdToOmittedParam(smart_objects::SmartObject& languages,
+                                const std::string& app_id) {
+    const size_t languages_array_size = languages.length();
+    for (size_t idx = 0; idx < languages_array_size; ++idx) {
+      smart_objects::SmartObject& language = languages[idx];
+      // ttsName verification
+      const std::string language_name = (*language.map_begin()).first;
+      if (!language[language_name].keyExists(json::ttsName)) {
+        language[language_name][json::ttsName] = app_id;
+        language.asString();
```

Comment:
@pold500 I think that this code line is redundant, maybe it was planned to be used as part of log message?